addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -706,8 +706,30 @@ export const editorCommands = {
             return selectedNodes.flatMap(node => {
                 let font = closestElement(node, 'font') || closestElement(node, 'span');
                 const children = font && descendants(font);
-                if (font && (font.nodeName === 'FONT' || (font.nodeName === 'SPAN' && font.style[mode]))) {
-                    // Partially selected <font>: split it.
+                const areAllChildrenSelected =
+                    children && children.every((child) => selectedNodes.includes(child));
+                const isGradient = font && isColorGradient(font.style["background-image"]);
+                const isTextGradient =
+                    isGradient && font.classList && font.classList.contains("text-gradient");
+                const shouldBreakGradient =
+                    (isTextGradient && mode === "color") ||
+                    (!isTextGradient && mode === "backgroundColor");
+
+                if (
+                    isGradient &&
+                    shouldBreakGradient &&
+                    !isColorGradient(color) &&
+                    areAllChildrenSelected
+                ) {
+                    font = [];
```

Comment:
There is already a else condition which sets `font=[]` can't we make the first two conditions false for your case?